### PR TITLE
RELATED: ONE-1734 Added support for using2 / inUseMany

### PR DESCRIPTION
**Pull request by BugsBunny338**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/232/ : SUCCESS in 3m 15s

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/236/ : SUCCESS in 1m 57s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/1/ : SUCCESS in 1m 30s (non-voting)

File: src/metadata.js [hunk 1]
```diff
@@ -73,6 +73,35 @@ export function getObjectUsing(projectId, uri, options = {}) {
...
+ * Get MD objects from using2 resource. Include only objects of given types
+ * and take care about fetching only nearest objects if requested.
+ *
+ * @method getObjectUsingMany
+ * @param {String} projectId id of the project
+ * @param {Array} uris uris of objects for which dependencies are to be found
+ * @param {Object} options objects with options:
+ *        - types {Array} array of strings with object types to be included
+ *        - nearest {Boolean} whether to include only nearest dependencies
+ * @return {Array} entries returned by using2 resource
```
Comment (BugsBunny338) [inline src/metadata.js]: Description edited.

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/237/ : SUCCESS in 2m 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/2/ : SUCCESS in 24s (non-voting)

File: test/metadata_test.js [hunk 2]
```diff
@@ -385,6 +385,100 @@ describe('metadata', () => {
...
+
+            beforeEach(() => {
+                postSpy = sinon.spy(xhr, 'post');
+            });
+
+            afterEach(() => {
+                postSpy.restore();
+            });
+
+            it('should load objects dependencies', done => {
```
Comment (akloboucnik) [inline test/metadata_test.js]: Leave out `done`, you can just return promise from the test and it will work

File: test/metadata_test.js [hunk 3]
```diff
@@ -385,6 +385,100 @@ describe('metadata', () => {
...
+                                nearest: 0
+                            }
+                        })
+                    })).to.be(true);
+                    expect(result).to.eql(response);
+                    done();
+                });
+            });
+
+            it('should be properly called with nearest when requested', done => {
```
Comment (akloboucnik) [inline test/metadata_test.js]: Same as before (leave out `done`)

File: test/metadata_test.js [hunk 4]
```diff
@@ -385,6 +385,100 @@ describe('metadata', () => {
...
+                                types,
+                                nearest: 1
+                            }
+                        })
+                    })).to.be(true);
+                    done();
+                });
+            });
+
+            it('should return rejected promise if 400 returned from backend', done => {
```
Comment (akloboucnik) [inline test/metadata_test.js]: Same here (leave out `done`)

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/238/ : SUCCESS in 2m 16s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/3/ : SUCCESS in 21s (non-voting)

**Pull request merged**